diff --git a/platform/unix/love.6 b/platform/unix/love.6
old mode 100644
new mode 100755
diff --git a/src/libraries/luasocket/libluasocket/auxiliar.c b/src/libraries/luasocket/libluasocket/auxiliar.c
index 95149701..cb00e7b4 100644
--- a/src/libraries/luasocket/libluasocket/auxiliar.c
+++ b/src/libraries/luasocket/libluasocket/auxiliar.c
@@ -24,7 +24,7 @@ int auxiliar_open(lua_State *L) {
 * Creates a new class with given methods
 * Methods whose names start with __ are passed directly to the metatable.
 \*-------------------------------------------------------------------------*/
-void auxiliar_newclass(lua_State *L, const char *classname, luaL_reg *func) {
+void auxiliar_newclass(lua_State *L, const char *classname, luaL_Reg *func) {
     luaL_newmetatable(L, classname); /* mt */
     /* create __index table to place methods */
     lua_pushstring(L, "__index");    /* mt,"__index" */
diff --git a/src/libraries/luasocket/libluasocket/auxiliar.h b/src/libraries/luasocket/libluasocket/auxiliar.h
index 18b84955..a4df9ebc 100644
--- a/src/libraries/luasocket/libluasocket/auxiliar.h
+++ b/src/libraries/luasocket/libluasocket/auxiliar.h
@@ -35,7 +35,7 @@
 #include "lauxlib.h"
 
 int auxiliar_open(lua_State *L);
-void auxiliar_newclass(lua_State *L, const char *classname, luaL_reg *func);
+void auxiliar_newclass(lua_State *L, const char *classname, luaL_Reg *func);
 void auxiliar_add2group(lua_State *L, const char *classname, const char *group);
 void auxiliar_setclass(lua_State *L, const char *classname, int objidx);
 void *auxiliar_checkclass(lua_State *L, const char *classname, int objidx);
diff --git a/src/libraries/luasocket/libluasocket/except.c b/src/libraries/luasocket/libluasocket/except.c
index c8dc5915..6d08608b 100644
--- a/src/libraries/luasocket/libluasocket/except.c
+++ b/src/libraries/luasocket/libluasocket/except.c
@@ -23,7 +23,7 @@ static int finalize(lua_State *L);
 static int do_nothing(lua_State *L);
 
 /* except functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"newtry",    global_newtry},
     {"protect",   global_protect},
     {NULL,        NULL}
diff --git a/src/libraries/luasocket/libluasocket/inet.c b/src/libraries/luasocket/libluasocket/inet.c
index dff84432..048c8824 100644
--- a/src/libraries/luasocket/libluasocket/inet.c
+++ b/src/libraries/luasocket/libluasocket/inet.c
@@ -23,7 +23,7 @@ static void inet_pushresolved(lua_State *L, struct hostent *hp);
 static int inet_global_gethostname(lua_State *L);
 
 /* DNS functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "toip", inet_global_toip },
     { "tohostname", inet_global_tohostname },
     { "gethostname", inet_global_gethostname},
diff --git a/src/libraries/luasocket/libluasocket/lua.h b/src/libraries/luasocket/libluasocket/lua.h
index a280df84..8731dcf6 100644
--- a/src/libraries/luasocket/libluasocket/lua.h
+++ b/src/libraries/luasocket/libluasocket/lua.h
@@ -7,7 +7,7 @@
 #include <lauxlib.h>
 
 #if LUA_VERSION_NUM > 501
-#	define luaL_reg luaL_Reg
+#	define luaL_Reg luaL_Reg
 #	define luaL_putchar(B, c) luaL_addchar(B, c)
 #	define luaL_typerror(L, n, t) luax_typerror(L, n, t)
 
diff --git a/src/libraries/luasocket/libluasocket/luasocket.c b/src/libraries/luasocket/libluasocket/luasocket.c
index a60090e7..35f5f935 100644
--- a/src/libraries/luasocket/libluasocket/luasocket.c
+++ b/src/libraries/luasocket/libluasocket/luasocket.c
@@ -49,7 +49,7 @@ static int base_open(lua_State *L);
 /*-------------------------------------------------------------------------*\
 * Modules and functions
 \*-------------------------------------------------------------------------*/
-static const luaL_reg mod[] = {
+static const luaL_Reg mod[] = {
     {"auxiliar", auxiliar_open},
     {"except", except_open},
     {"timeout", timeout_open},
@@ -61,7 +61,7 @@ static const luaL_reg mod[] = {
     {NULL, NULL}
 };
 
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"skip",      global_skip},
     {"__unload",  global_unload},
     {NULL,        NULL}
diff --git a/src/libraries/luasocket/libluasocket/mime.c b/src/libraries/luasocket/libluasocket/mime.c
index 66098911..88f1b172 100644
--- a/src/libraries/luasocket/libluasocket/mime.c
+++ b/src/libraries/luasocket/libluasocket/mime.c
@@ -50,7 +50,7 @@ static size_t qpencode(UC c, UC *input, size_t size,
 static size_t qppad(UC *input, size_t size, luaL_Buffer *buffer);
 
 /* code support functions */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "dot", mime_global_dot },
     { "b64", mime_global_b64 },
     { "eol", mime_global_eol },
diff --git a/src/libraries/luasocket/libluasocket/select.c b/src/libraries/luasocket/libluasocket/select.c
index 73848ce2..35185e25 100644
--- a/src/libraries/luasocket/libluasocket/select.c
+++ b/src/libraries/luasocket/libluasocket/select.c
@@ -29,7 +29,7 @@ static void make_assoc(lua_State *L, int tab);
 static int global_select(lua_State *L);
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"select", global_select},
     {NULL,     NULL}
 };
diff --git a/src/libraries/luasocket/libluasocket/tcp.c b/src/libraries/luasocket/libluasocket/tcp.c
index 7e26b080..38aef089 100644
--- a/src/libraries/luasocket/libluasocket/tcp.c
+++ b/src/libraries/luasocket/libluasocket/tcp.c
@@ -40,7 +40,7 @@ static int meth_setfd(lua_State *L);
 static int meth_dirty(lua_State *L);
 
 /* tcp object methods */
-static luaL_reg tcp[] = {
+static luaL_Reg tcp[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"accept",      meth_accept},
@@ -75,7 +75,7 @@ static t_opt opt[] = {
 };
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"tcp", global_create},
     {NULL, NULL}
 };
diff --git a/src/libraries/luasocket/libluasocket/timeout.c b/src/libraries/luasocket/libluasocket/timeout.c
index 8db087f4..65fb75d5 100644
--- a/src/libraries/luasocket/libluasocket/timeout.c
+++ b/src/libraries/luasocket/libluasocket/timeout.c
@@ -35,7 +35,7 @@ extern void luax_register(lua_State *L, const char *name, const luaL_Reg *l);
 static int timeout_lua_gettime(lua_State *L);
 static int timeout_lua_sleep(lua_State *L);
 
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     { "gettime", timeout_lua_gettime },
     { "sleep", timeout_lua_sleep },
     { NULL, NULL }
diff --git a/src/libraries/luasocket/libluasocket/udp.c b/src/libraries/luasocket/libluasocket/udp.c
index 640c459e..141517e6 100644
--- a/src/libraries/luasocket/libluasocket/udp.c
+++ b/src/libraries/luasocket/libluasocket/udp.c
@@ -45,7 +45,7 @@ static int meth_setfd(lua_State *L);
 static int meth_dirty(lua_State *L);
 
 /* udp object methods */
-static luaL_reg udp[] = {
+static luaL_Reg udp[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"close",       meth_close},
@@ -78,7 +78,7 @@ static t_opt opt[] = {
 };
 
 /* functions in library namespace */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"udp", global_create},
     {NULL, NULL}
 };
diff --git a/src/libraries/luasocket/libluasocket/unix.c b/src/libraries/luasocket/libluasocket/unix.c
index fb12903d..32719168 100644
--- a/src/libraries/luasocket/libluasocket/unix.c
+++ b/src/libraries/luasocket/libluasocket/unix.c
@@ -42,7 +42,7 @@ static const char *unix_tryconnect(p_unix un, const char *path);
 static const char *unix_trybind(p_unix un, const char *path);
 
 /* unix object methods */
-static luaL_reg un[] = {
+static luaL_Reg un[] = {
     {"__gc",        meth_close},
     {"__tostring",  auxiliar_tostring},
     {"accept",      meth_accept},
@@ -74,7 +74,7 @@ static t_opt opt[] = {
 };
 
 /* our socket creation function */
-static luaL_reg func[] = {
+static luaL_Reg func[] = {
     {"unix", global_create},
     {NULL,          NULL}
 };
